### PR TITLE
[RSDK-9216] - allocate buffer for response message per RPC call

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

**Comment by gvaradarajan** [comment]
still need to test on an actual ESP32 with camera

File: micro-rdk/src/common/grpc.rs [hunk 1]
```diff
@@ -836,7 +822,9 @@ impl<'a> GrpcServerInner<'a> {
             .get_generic_readings()
             .map_err(|err| ServerError::new(GrpcError::RpcInternal, Some(err.into())))?;
         let resp = proto::common::v1::GetReadingsResponse { readings };
-        self.encode_message(resp)
+        let res = GrpcServerInner::get_encoded_message(resp);
+        println!("res: {:?}", res);
```
Comment (acmorrow) [inline micro-rdk/src/common/grpc.rs:826]: Looks like a stale debugging print here.

File: micro-rdk/src/common/grpc.rs [hunk 2]
```diff
@@ -154,28 +148,20 @@ where
 }
 
 impl<'a> GrpcServerInner<'a> {
-    fn encode_message_into<M: Message>(m: M, mut buffer: BytesMut) -> Result<Bytes, ServerError> {
+    fn get_encoded_message<M: Message>(m: M) -> Result<Bytes, ServerError> {
```
Comment (acmorrow) [inline micro-rdk/src/common/grpc.rs:151]: I think I like the old name better. This method isn't getting an encoded message from somewhere, it is taking a message and encoding it.

Since you no longer need the "new buffer" vs "internal buffer" distinction with the three existing methods were intended to support and now only need the former, you could just call this method `encode_message`.

File: micro-rdk/src/common/camera/fake_camera.rs [hunk 3]
```diff
@@ -130,25 +111,26 @@ mod tests {
...
+        let listener: async_io::Async<TcpListener> = tcp_server.try_into().unwrap();
 
         let robot = Arc::new(Mutex::new(setup_robot().unwrap()));
 
         loop {
             let incoming = listener.accept().await;
             assert!(incoming.is_ok());
-            let stream = incoming.unwrap();
+            let incoming = incoming.unwrap();
+            let stream: NativeStream = NativeStream::LocalPlain(incoming.0);
```
Comment (acmorrow) [inline micro-rdk/src/common/camera/fake_camera.rs:133]: What is the context for these changes? They seem somewhat unrelated to the rest?

File: micro-rdk/src/esp32/camera.rs [hunk 4]
```diff
@@ -173,36 +173,9 @@ impl Esp32Camera {
...
-        let msg = HttpBody {
-            content_type: "image/jpeg".to_string(),
-            data: frame.as_bytes().to_vec(),
-            ..Default::default()
-        };
-        // safety: message must be encoded before the frame is dropped from scope
-        msg.encode(&mut buffer)
-            .map_err(CameraError::MessageEncodeError)?;
-        return Ok(buffer);
+        Ok(frame.as_bytes())
```
Comment (acmorrow) [inline micro-rdk/src/esp32/camera.rs:178]: Does this allocate?

**Review by acmorrow — CHANGES_REQUESTED** [review]

File: micro-rdk/src/common/camera/fake_camera.rs (hunk 3 above)
Comment (gvaradarajan) [reply micro-rdk/src/common/camera/fake_camera.rs:133]: the camera tests don't pass without it, ever since [this commit](https://github.com/viamrobotics/micro-rdk/commit/b4f543140e2bb919d7294c97352f35d69b27dbed)

**Review by mattjperez — APPROVED** [review]
Pretty floored that the solution was so straight forward. 
LGTM, verified worki with camera

**Review by acmorrow — APPROVED** [review]
LGTM
